internal/agent/install.go

```diff
@@ -102,11 +102,13 @@ func Install(dir ...string) error {
 		}
 	})
 
-	// Try to pull userdata once more. best-effort
-	if _, err := os.Stat("/oem/userdata"); err != nil {
-		if err := machine.ExecuteCloudConfig("/system/oem/00_datasource.yaml", "rootfs.before"); err != nil {
-			fmt.Println("Warning: Failed pulling from datasources")
+	for {
+		if _, err := os.Stat("/oem/.userdata_load"); os.IsNotExist(err) {
+			break
 		}
+
+		fmt.Println("userdata configuration has not yet completed.")
```

Comment:
maybe we should write down what we are waiting for (for debugging reasons)

```suggestion
		fmt.Println("userdata configuration has not yet completed (waiting /oem/.userdata_load) .")
```

Reply:
I like the idea of more detail, ill give some context to what that file means as well.